instrumentation/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -179,6 +184,24 @@ public OpenTelemetryRumBuilder addPropagatorCustomizer(
         return this;
     }
 
+    /**
+     * Adds a {@link Function} to invoke with the default {@link SpanExporter} to allow
+     * customization. The return value of the {@link Function} will replace the passed-in argument.
+     *
+     * <p>Multiple calls will execute the customizers in order.
+     */
+    public OpenTelemetryRumBuilder addSpanExporterCustomizer(
```

Comment:
ah, I think you want `addTracerProviderCustomizer()` instead for the use case you are targeting

`addSpanExporterCustomizer()` is for wrapping (customizing) an existing span exporter

Reply:
@trask - Apparently there is more than one way to skin this cat 🐱 and I'm sure Guido would disapprove. 😁 

For what it's worth, the `addTracerProviderCustomizer()` has already been in place, and we use it all over the place in our distro, including [setting up the zipkin-based exporter](https://github.com/signalfx/splunk-otel-android/blob/main/splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java#L132).  For most users who just want to plug in their own (preconfigured) exporter, they probably don't want to deal with span processors -- they just want to tell the android sdk to use their exporter. Period.

In any case, there's now closer parity with the autoinstrumentation builder stuff, we have both `addTracerProviderCustomizer()` and `addExporterCustomizer()` in both the core autoinstrumentation and in android now.

Reply:
> For most users who just want to plug in their own (preconfigured) exporter, they probably don't want to deal with span processors -- they just want to tell the android sdk to use their exporter. Period.

outside of android we'd recommend `ConfigurableSpanExporterProvider` for this use case. I forget, does android not support SPI?

maybe we could have a new non-SPI way to register our normal SPI providers in RumBuilder, e.g. `RumBuilder.registerProvider()`?

Reply:
> does android not support SPI?

It does, but it was broken at some point and I'm also not sure if the common practice of [obfuscating the code](https://developer.android.com/build/shrink-code#obfuscate) might cause some issues too, issues that we could avoid by setting proguard consumer rules to prevent obfuscation for some classes, though in general it seems like using SPIs should be done carefully in android, so I would say it's good to know that we can use them if needed which is nice, but I would prefer to use them only when there's no better alternatives.

Reply:
Yeah, what he said. ^^ 😁 

We also don't currently take a dependency on the core autoconfigure sdk.